src/realm/object-store/impl/realm_coordinator.cpp

```
@@ -309,6 +309,19 @@ ThreadSafeReference RealmCoordinator::get_unbound_realm()
 
 void RealmCoordinator::do_get_realm(Realm::Config config, std::shared_ptr<Realm>& realm,
                                     util::Optional<VersionID> version, util::CheckedUniqueLock& realm_lock)
+{
+    try {
+        open_and_init_realm(config, realm, version, realm_lock);
+    }
+    catch (const std::exception& e) {
+        if (realm)
```

I need to re-throw the exception..